.github/workflows/gradle-publish.yml

```diff
@@ -0,0 +1,44 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# This workflow will build a package using Gradle and then publish it to GitHub packages when a release is created
+# For more information see: https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#Publishing-using-gradle
+
+name: Gradle Package
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up JDK 17
+      uses: actions/setup-java@v4
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+        server-id: github # Value of the distributionManagement/repository/id field of the pom.xml
+        settings-path: ${{ github.workspace }} # location for the settings.xml file
+
+    - name: Setup Gradle
+      uses: gradle/actions/setup-gradle@417ae3ccd767c252f5661f1ace9f835f9654f2b5 # v3.1.0
+
+    - name: Build with Gradle
+      run: ./gradlew build
+
+    # The USERNAME and TOKEN need to correspond to the credentials environment variables used in
+    # the publishing section of your build.gradle
+    - name: Publish to GitHub Packages
+      run: ./gradlew publish
+      env:
+        USERNAME: ${{ github.actor }}
+        TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
**Optimization Issue**: The workflow for publishing the package to GitHub packages upon release creation could benefit from optimization in terms of security and efficiency. Using third-party actions poses a potential risk if not properly managed and versioned. Additionally, the build and publish steps could be optimized to only trigger when necessary, reducing unnecessary runs and potential errors. <br> **Fix**: - Pin the third-party actions to a specific commit SHA to mitigate the risk of malicious code injection from compromised or updated actions. For example, change 'uses: actions/checkout@v4' to 'uses: actions/checkout@<commit-sha>'.
- Add conditional steps or jobs to ensure that the build and publish steps only run when there are meaningful changes to the package, thus optimizing CI/CD resources. This can be achieved by using 'paths' or 'paths-ignore' options in the workflow trigger section or by adding conditional checks within the job. <br> **Code Suggestion**: 
 ```
 - uses: actions/checkout@<commit-sha>
   if: github.event_name == 'release' && github.event.action == 'created'
 - uses: actions/setup-java@<commit-sha>
   with:
     java-version: '17'
     distribution: 'temurin'
     server-id: github
     settings-path: ${{ github.workspace }}
   if: github.event_name == 'release' && github.event.action == 'created'
 ```